andela/ah-the-jedi-backend
# #165305282 Comments should be able to track edit history.

#### What does this PR do?
- Implement comments should be able to track edit history
#### Description of Task to be completed?
- Implement update comment endpoint
- Create a model to save the previous comment
- Get all comment edit history
#### How should this be manually tested?
1. Clone this repo.
   `$ git clone https://github.com/andela/ah-the-jedi-backend.git`
2. Fetch all origin to get all branches
   `$ git fetch origin`
3. Checkout to `ft-comment-edit-history-165305282` branch.
   `$ git checkout ft-comment-edit-history-165305282`
4. Activate `virtualenv`
    `$ virtualenv -p python .venv`
    `$ source .venv/bin/activate`
5. Install requirements
   `$ pip install -r requirements.txt`
6. Start server
   `$ python manage.py runserver`
7. Test the endpoint on postman in the following steps:
- Ensure the user is authenticated.
- Update a comment - `PUT /api/articles/<slug>/comments/?id=`
- Track comment edit history by getting previous comments `GET /api/articles/<slug>/comments/<comment_id>/history/`
#### What are the relevant pivotal tracker stories?
[#165305282](https://www.pivotaltracker.com/story/show/165305282)
#### Screenshot
Successfully update comment
<img width="1102" alt="Screenshot 2019-05-15 at 16 59 34" src="https://user-images.githubusercontent.com/17095461/57781567-22a84000-7733-11e9-8a47-3bc75fea5ecb.png">
<img width="1106" alt="Screenshot 2019-05-16 at 09 51 52" src="https://user-images.githubusercontent.com/17095461/57832468-6a28dd80-77c0-11e9-8b78-6d5c54b57765.png">


Track comment edit history
<img width="1103" alt="Screenshot 2019-05-16 at 09 51 19" src="https://user-images.githubusercontent.com/17095461/57832476-6eed9180-77c0-11e9-9b91-87aea68ed5ba.png">


User cannot update a comment they do not own
<img width="1099" alt="Screenshot 2019-05-15 at 16 58 41" src="https://user-images.githubusercontent.com/17095461/57781613-3b185a80-7733-11e9-8b3b-2870e95cae93.png">
